src/internal/operators/throttle.ts

```diff
@@ -63,92 +63,43 @@ export const defaultThrottleConfig: ThrottleConfig = {
-                            config: ThrottleConfig = defaultThrottleConfig): MonoTypeOperatorFunction<T> {
-  return (source: Observable<T>) => lift(source, new ThrottleOperator(durationSelector, !!config.leading, !!config.trailing));
-}
-
-class ThrottleOperator<T> implements Operator<T, T> {
-  constructor(private durationSelector: (value: T) => SubscribableOrPromise<any>,
-              private leading: boolean,
-              private trailing: boolean) {
-  }
-
-  call(subscriber: Subscriber<T>, source: any): TeardownLogic {
-    return source.subscribe(
-      new ThrottleSubscriber(subscriber, this.durationSelector, this.leading, this.trailing)
-    );
-  }
-}
-
-/**
- * We need this JSDoc comment for affecting ESDoc
- * @ignore
- * @extends {Ignored}
- */
-class ThrottleSubscriber<T, R> extends SimpleOuterSubscriber<T, R> {
-  private _throttled: Subscription | null | undefined;
-  private _sendValue: T | null = null;
-  private _hasValue = false;
-
-  constructor(protected destination: Subscriber<T>,
-              private durationSelector: (value: T) => SubscribableOrPromise<number>,
-              private _leading: boolean,
-              private _trailing: boolean) {
-    super(destination);
-  }
-
-  protected _next(value: T): void {
-    this._hasValue = true;
-    this._sendValue = value;
-
-    if (!this._throttled) {
-      if (this._leading) {
-        this.send();
-      } else {
-        this.throttle(value);
-      }
-    }
-  }
-
-  private send() {
-    const { _hasValue, _sendValue } = this;
-    if (_hasValue) {
-      this.destination.next(_sendValue!);
-      this.throttle(_sendValue!);
-    }
-    this._hasValue = false;
-    this._sendValue = null;
-  }
-
-  private throttle(value: T): void {
-    let result: SubscribableOrPromise<any>;
-    try {
-      result = this.durationSelector(value);
-    } catch (err) {
-      this.destination.error(err);
-      return
-    }
-    this.add(this._throttled = innerSubscribe(result, new SimpleInnerSubscriber(this)));
-  }
-
-  private throttlingDone() {
-    const { _throttled, _trailing } = this;
-    if (_throttled) {
-      _throttled.unsubscribe();
-    }
-    this._throttled = null;
-
-    if (_trailing) {
-      this.send();
-    }
-  }
-
-  notifyNext(): void {
-    this.throttlingDone();
-  }
-
-  notifyComplete(): void {
-    this.throttlingDone();
-  }
+export function throttle<T>(
+  durationSelector: (value: T) => SubscribableOrPromise<any>,
+  { leading, trailing }: ThrottleConfig = defaultThrottleConfig
+): MonoTypeOperatorFunction<T> {
+  return (source: Observable<T>) =>
+    lift(source, function (this: Subscriber<T>, source: Observable<T>) {
+      const subscriber = this;
+      let hasValue = false;
+      let sendValue: T | null = null;
+      let throttled: Subscription | null = null;
+
+      const throttlingDone = () => {
+        throttled?.unsubscribe();
+        throttled = null;
+        trailing && send();
```

Comment:
issue: I think this should also include `!subscriber.closed` in the condition - `trailing && !subscriber.closed && send()` - as `throttlingDone` is called upon unsubscription and it calls `send()`, which subscribes to the duration selector and that's pointless if the subscription has been torn down.

Reply:
you'll notice that `!this.closed` is checked before this unsubscribe handler is called.